### PR TITLE
RUMM-1397 Update Endpoints

### PR DESCRIPTION
**Pull request by xgouchet**
### What and why?

Update the endpoints available in the SDK and mark previous ones as deprecated.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: Sources/DatadogObjc/DatadogConfiguration+objc.swift [hunk 1]
```diff
@@ -16,6 +16,18 @@ public class DDEndpoint: NSObject {
...
+    public static func us1() -> DDEndpoint { .init(sdkEndpoint: .us1) }
+    
+    @objc
+    public static func us3() -> DDEndpoint { .init(sdkEndpoint: .us3) }
+    
+    @objc
+    public static func eu1() -> DDEndpoint { .init(sdkEndpoint: .eu1) }
+    
+    @objc
+    public static func us1_fed() -> DDEndpoint { .init(sdkEndpoint: .us1_fed) }
```
Comment (buranmert) [inline Sources/DatadogObjc/DatadogConfiguration+objc.swift:30]: we have `DDLogsEndpoint` / `DDTracesEndpoint` but we don't have `DDRUMEndpoint`, did we forget it or should we deprecate `DDLogsEndpoint` / `DDTracesEndpoint` as well?

**Review by buranmert — APPROVED** [review]
lgtm but i left a question regarding `DDRUMEndpoint`

File: Sources/DatadogObjc/DatadogConfiguration+objc.swift (hunk 1 above)
Comment (ncreated) [reply Sources/DatadogObjc/DatadogConfiguration+objc.swift:30]: Both `set(logsEndpoint: DDLogsEndpoint)` and `public func set(tracessEndpoint: DDLogsEndpoint)` were deprecated when introducing common `func set(endpoint: DDEndpoint)`. Enum cases were not deprecated, but they could be.

**Review by ncreated — APPROVED** [review]
LG 👍

**Pull request merged**